decision_service/include/object_pool.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+#include <mutex>
+#include <vector>
+
+namespace reinforcement_learning {
+  template<typename TObject>
+  class pooled_object {
+  private:
+    TObject * _val;
+
+  public:
+    pooled_object(TObject* obj, int pversion)
+      : _val(obj), version(pversion)
+    { }
+
+    ~pooled_object()
+    {
+      if (_val) {
+        delete _val;
```

Comment:
>delete _val; [](start = 8, length = 12)

need to either a) delete copy const, assignment + moves or b) implement them to prevent accidental misuse